src/plugins/console/public/lib/autocomplete/autocomplete.ts

```diff
@@ -349,14 +349,66 @@ export default function ({
     });
   }
 
+  function hasOneOf(value: unknown): value is { __one_of: ConditionalTemplateType[] } {
```

Comment:
As someone who isn't as familiar with Console and the autocomplete logic, I'd find it really helpful to add comments explaining each of these functions added.